zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -377,28 +376,6 @@ public void parseProperties(Properties zkProp) throws IOException, ConfigExcepti
                 backupConfigBuilder.setTimetableStoragePath(value);
             } else if (key.equals(BackupSystemProperty.BACKUP_TIMETABLE_BACKUP_INTERVAL_MS)) {
                 backupConfigBuilder.setTimetableBackupIntervalInMs(Long.parseLong(value));
-            } else if (key.equals(X509AuthenticationConfig.SET_X509_CLIENT_ID_AS_ACL)) {
```

Comment:
If this is not correct then essentially we are doing lazy binding of these variables. Other way to fix this is fix these if blocks. I would suggest separate out this change into separate PR and test it independently and make sure "all" these fields are lazily bound in their getter methods.